### ISSUE TITLE
SplashActivity does not excluded from CLEVERTAP_INAPP_EXCLUDE

### ISSUE THREAD

**Issue by afarhan39** [CleverTap/clevertap-android-sdk#218]
**Describe the bug**
In App Notifications does not obey CLEVERTAP_INAPP_EXCLUDE tag despite added correct Activity to [exclude](https://docs.clevertap.com/docs/faq#q-why-does-the-in-app-notification-disappear-from-the-users-device-in-a-few-seconds
).

**To Reproduce**
1. Add SplashActivity inside CLEVERTAP_INAPP_EXCLUDE
2. Create campaign for in app notification, and send test campaign
3. Received push notification due to send test campaign, refer [here](https://docs.clevertap.com/docs/faq#q-why-do-i-receive-a-clevertap-push-notification-even-after-sending-a-test-in-app)
4. Tap on push notification. App will open and showing In App Notification on Splash, and dismissed it when going to HomeActivity, despite already done Step 1.

**Expected behavior**
I'm expecting In App Notification to be shown in HomeActivity, instead of SplashActivity.

**Screenshots/Logs**
Please see attached video. [InAppVid.webm
](https://drive.google.com/file/d/1sV1FW9AmAplbFuosPRthw4-H_Hf9YZ4W/view)

**Environment (please complete the following information):**
 - Device: Android Emulator
 - OS: Android 12
 - CleverTap SDK Version: 4.4.0
 - Android Studio Version 2020.3.1 Patch 4

**Additional context**
- Activity is written in Kotlin
- Had to use Emulator due to some reason my device does not able to receive any push notification for in app. Might open a new issue for this.

**Comment by root-ansh-ct** [comment]
Hi @afarhan39  . can you please do the following:
1.  provide logs till the login screen, filtered by "clevertap" tag
2. verify if you added the whole name of the activity(i.e package+activity name) or just the class name, in the manifest?
3.  share the conditions used for creating the target audience Mainly the values here, when you create the campaign: 
![image](https://user-images.githubusercontent.com/95351764/149724172-b7932a0b-0a71-4ad1-880d-aee86650b379.png)

**Comment by william-ct** [comment]
@walaankit To exclude in-app from any specified activity, you only need to specify only the activity name. More info regarding the same can be [found here](https://developer.clevertap.com/docs/android-in-app-notifications#exclude-in-app-from-android-activity)

**Comment by AkimovAN** [comment]
@william-ct I spent a lot of time debugging and found the following - you need to specify the full name of the activity 
![изображение](https://user-images.githubusercontent.com/43830473/225852216-4df76230-58b5-4267-a1db-1b9880426f61.png)

[this method in library](https://github.com/CleverTap/clevertap-android-sdk/blob/2c1feb1ff93f349490d2e6165d93365efb6bde00/clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/InAppController.java#L471)

**Comment by william-ct** [comment]
@AkimovAN 
> you need to specify the full name of the activity

How did you conclude the above result from the sample you've shared? In the screenshot, you're blacklisted activity is `StartActivity` and your current activity `ui.activities.SplashScreenActivity`, in this specific case `canShowInAppOnActivity()` will return true.

Also if you can see line 343 we get the local class name from `CoreMetaData.getCurrentActivityName()` and does a **contains** check with `blackListedActivity` string value. So if it(`blackListedActivity`) has the entire ActivityName/getLocalClassName(package_activityName) or only the activityName/getSimpleClassName(activityName), `canShowInAppOnActivity()` method  will return false when `currentActivityName` contains `blackListedActivity` string value.

@walaankit Do let us know here if it has solved your issue.Thanks

**Comment by AkimovAN** [comment]
@william-ct 
Ignore the activity names from the screenshot - I only show that if you specify only one activity name in the manifest, then this name will be added to the inappActivityExclude list, while CoreMetaData.getCurrentActivityName () returns the full name of the activity, including package name. This is what I was trying to show in the screenshot

**Comment by william-ct** [comment]
@AkimovAN So if you specify only activityName or activityName(including the package) in the manifest, then in-app notification will always be excluded/not shown in the specified activity.

**Comment by AkimovAN** [comment]
@william-ct 
No, that's what I'm talking about. If you add  the only activity name ("TestActivity") to the manifest, then the list (inappActivityExclude) will store "TestActivity", while CoreMetaData.getCurrentActivityName() will return "com.your.company.name.TestActivity" and check will fail

**Comment by AkimovAN** [comment]
@william-ct 
Sorry, I mistyped the list name a bit. Of course, we were talking about the "blackListedActivity" list. But everything else is correct. I debugged it again

![изображение](https://user-images.githubusercontent.com/43830473/225933159-969f15a4-2b44-427f-aa4a-8637530e4953.png)

**Comment by AkimovAN** [comment]
@william-ct 
You are absolutely right, I understood what was wrong, thank you very much for your answers
